flake.nix

```diff
@@ -0,0 +1,88 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      name = "liboqs";
+      version = "0.0.1";
```

Comment:
I think maintaining two versions of `liboqs` and `flake.nix` are confusing. If this version of `liboqs` introduce `flake.nix`, let's start at the same version with `liboqs`. Thought? @baentsch  @SWilson4 .

Reply:
Sounds good - I'll see if I can build it without having a version or have the version inherit from CMake.